### ISSUE TITLE
Error trying to set upscale_limit to zero

### ISSUE THREAD

**Issue by zenput-francisco** [carwow/terraform-provider-hirefire#18]
Hi. We're using your provider to handle our Hirefire resources and we recently tried to set `upscale_timeout` to zero and that resulted in:
```
Error: 422: {"errors":{"upscale_timeout":["can't be blank","is not a number"]}}
 
   with hirefire_manager.highpriority,
   on hirefire.tf line 22, in resource "hirefire_manager" "highpriority":
   22: resource "hirefire_manager" "highpriority" {
```
The value is valid in the UI but fails with the provider. We're using Terraform `1.0.11` and version `0.4.0` of the carwow Hirefire provider.

Here's the manager definition:
```
resource "hirefire_manager" "highpriority" {
  application_id        = hirefire_application.application.id
  name                  = "highpriority"
  type                  = "Manager::Worker::HireFire::JobQueue"
  enabled               = true
  minimum               = 0
  maximum               = 2
  notify                = false
  notify_quantity       = 1
  notify_after          = 1
  ratio                 = 250
  decrementable         = true
  upscale_sensitivity   = 1
  downscale_sensitivity = 1
  upscale_timeout       = 0
  downscale_timeout     = 1
  upscale_limit         = 1
  downscale_limit       = 0
}
```
Let me know if you need any other details. And thank you for your help.

**Comment by emancu** [comment]
Thanks @zenput-francisco for reporting this, we are investigating what could be the issue. We will keep you posted.

**Comment by zenput-francisco** [comment]
Thank you!
Another update just had the same problem with `upscale_limit` and `downscale_limit` set to zero which in the above example didn't had any issues:

```
│ Error: 422: {"errors":{"upscale_limit":["can't be blank","is not a number"],"downscale_limit":["can't be blank","is not a number"]}}
│ 
│   with hirefire_manager.production-taskqueue,
│   on [production-managers.tf](http://production-managers.tf/) line 82, in resource "hirefire_manager" "production-taskqueue":
│   82: resource "hirefire_manager" "production-taskqueue" {
```
And here's the full resource definition:
```
resource "hirefire_manager" "production-taskqueue" {
  application_id        = hirefire_application.production-zenput.id
  name                  = "taskqueue"
  type                  = "Manager::Worker::HireFire::JobQueue"
  enabled               = true
  minimum               = 0
  maximum               = 1
  notify                = false
  notify_quantity       = 1
  notify_after          = 1
  ratio                 = 16
  decrementable         = true
  upscale_sensitivity   = 1
  downscale_sensitivity = 1
  upscale_timeout       = 1
  downscale_timeout     = 5
  upscale_limit         = 1
  downscale_limit       = 1
}
```
If you need more details let me know. Thank your for your help.

**Comment by PChambino** [comment]
Hi @zenput-francisco, thank you for reporting.
The Hirefire API had subtle changes and we also had the same issue internally.
We have released a new version v0.4.1 which should fix the zero/null values not being applied.
Let us know if it works for you or if there are any other issues.
